Add NothingPersonalHomePage styling

ramram-mf opened the pull request:
# Description

This PR implements styling for the Nothing Personal Homepage in all viewports.

In order to play Vimeo videos on the `featured_items` section the `CSP_MEDIA_SRC` environment variable should be updated to add `https://player.vimeo.com/ https://download-video-ak.vimeocdn.com` to its value. The `env.default` file reflects this update in order to allow new development environments to include the mentioned hosts.

![nothing-personal-homepage-stacked-sections](https://github.com/user-attachments/assets/80cde17a-5c21-4785-963c-f062ae2929af)
 
*Nothing Personal Homepage `large` viewport*

Link to sample test page: https://foundation-s-tp1-3158-n-h9g1fi.herokuapp.com/en/nothing-personal/
Link to UI regression test page: https://foundation-s-tp1-3158-n-h9g1fi.herokuapp.com/en/ui-sandbox/
Credentials: `admin` / `3#NXfj105^BW8ZEm`
Related PRs/issues: [Jira Ticket](https://mozilla-hub.atlassian.net/browse/TP1-3158)

mmmavis commented on foundation_cms/static/scss/type.scss:
```diff
@@ -394,3 +394,32 @@ body.redesign {
   @include mofo-text-style($link-text-styles, "regular", $header-font-family);
 }
+
+// ===================================
+// Highlight effect
+// ===================================
+
+@mixin highlight-effect($highlight-color, $highlight-gap) {
+  display: inline;
+  box-decoration-break: clone;
+
+  // The highlight effect is created using a linear gradient background image
+  // with transparent slices at the top and bottom, and a solid color in the middle.
+  // Using a background color alone would not allow for the gap between lines
+  // of text in multi-line headers.
+  background-image: linear-gradient(
+    to bottom,
+    /* top transparent slice */ transparent 0,
+    transparent calc(#{$highlight-gap} / 2),
+    /* highlight body */ $highlight-color calc(#{$highlight-gap} / 2),
+    $highlight-color calc(100% - #{$highlight-gap} / 2),
+    /* bottom transparent slice */ transparent
+      calc(100% - #{$highlight-gap} / 2),
+    transparent 100%
+  );
+  background-size: 100%;
+  background-position: 0 1px;
+  background-repeat: repeat-y;
+  background-origin: padding-box;
+  background-clip: padding-box;
+}
```
Thanks for turning this into a mixin. I'd like `type.scss` to focus on base type styles (font size, family, weight, etc.). Could you create a new `_mixins.scss` file and move this mixin there? (I'm doing the same [in my to-be-reviewed Product Review Page PR](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/14655/files#diff-9bdf2256b390c5c91cba3c6ebc1fe5d1660868befc0f40259ffb7ac5867436a6)) It would also help to rename the mixin to something more specific.

mmmavis commented on foundation_cms/templates/patterns/blocks/themes/default/text_image_block__narrow.html:
```diff
@@ -1,3 +1,34 @@
-{% extends "./text_image_block.html" %}
```
Similar concern --

Quite a few changes were made to this file. If they're intentional for this PR, could you please set up a review app page (outside of NP) so we can confirm there are no regressions?

ramram-mf commented on env.default:
```diff
@@ -54,7 +54,7 @@ CSP_FONT_SRC=" 'self' data: https://code.cdn.mozilla.net https://*.fundraiseup.c
 CSP_FRAME_SRC=" 'self' https://www.youtube.com https://comments.mozillafoundation.org/ https://airtable.com https://docs.google.com/ https://platform.twitter.com https://public.zenkit.com https://calendar.google.com https://www.youtube-nocookie.com https://form.typeform.com https://js.tito.io https://datawrapper.dwcdn.net https://www.google.com/recaptcha/ https://pay.google.com https://*.paypal.c ...
 CSP_IMG_SRC=" * data: blob: https://*.fundraiseup.com https://ucarecdn.com https://pay.google.com https://*.paypalobjects.com "
-CSP_MEDIA_SRC=" 'self' data: https://s3.amazonaws.com/mofo-assets/foundation/video/ "
+CSP_MEDIA_SRC=" 'self' data: https://s3.amazonaws.com/mofo-assets/foundation/video/ https://player.vimeo.com/ https://download-video-ak.vimeocdn.com "
```
It looks like the continuous integration one was missing entirely this variable, adding it as maintenance.

ramram-mf commented:
@nancyt1 

>     * On mobile, there's not a lot of room to click this link before the next one shows up, I'm wondering if we should disable this sticky feature on mobile?

Since the effect looks quite nice, I tried increasing the section overall height by a 15% and shifting content 20% up so that we have more both scroll time due to length and clickable window due to position. Lets test it and see how it works. If this still falls short of the desired experience we can always disable the stacking effect in mobile.

All other feedback is also addressed for review.

@mmmavis 

Thanks for the feedback, updated all of it. Related to the regressions, you're right I actually set that up locally and now there is review app ui regression test included in the PR description so that we can check quickly if something broke with the external component refactor. This refactor was specifically done to isolate any existing styling to the text-image block, which previously only existed inside the tabbed content container. So I set up one container with the block to check it remains in place.

The `mixins.scss` file had to spare the underscore due to our linter rules :(

<img width="1169" height="935" alt="image" src="https://github.com/user-attachments/assets/76081006-3953-4734-a88e-ca6db66dd0b5" />

ramram-mf commented on foundation_cms/static/scss/blocks/themes/default/tabbed_content_container.scss:
```diff
@@ -123,87 +123,87 @@
       }
     }
   }
-}
```
It looks like there was a lot of updating due to gits diffing methods, but the only thing that changed was that now the text-image block styles are nested inside the container rule for scoping. Shifted a few lines up inside the }

kevdmoz commented:
Hi @ramram-mf still seeing these accessibility warnings - let me know when you get to them and I can re-check - cheers

> @ramram-mf looking good so far! leaving some accessibility notes:
> 
> **Mobile** <img alt="image" width="1197" height="445" src="https://private-user-images.githubusercontent.com/200307766/494490316-b23012b9-69db-4ffe-9d5b-f07c39a6ee32.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkxNjU0NzQsIm5iZiI6MTc1OTE2NTE3NCwicGF0aCI6Ii8yMDAzMDc3NjYvNDk0NDkwMzE2LWIyMzAxMmI5LTY5ZGItNGZmZS05ZDViLWYwN2MzOWE2ZWUzMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyOVQxNjU5MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NWRhZTM0YmQ1NDRhOTQyZTI1OThjMmRkNGU4OGRjZjRmM2NiNDAwMDIxNDExODBhM2I4Y2Q1OWMwNTRhNzU3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1Sy7lJmpwZO3owj10TLYiL6O11rXs0slOYTq9LuGBsY">
> 
> checking if we have fields to add alt text to images for product review images <img alt="image" width="1189" height="524" src="https://private-user-images.githubusercontent.com/200307766/494490533-1aa868e2-b777-41b1-8f91-1bb427b04723.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkxNjU0NzQsIm5iZiI6MTc1OTE2NTE3NCwicGF0aCI6Ii8yMDAzMDc3NjYvNDk0NDkwNTMzLTFhYTg2OGUyLWI3NzctNDFiMS04ZjkxLTFiYjQyN2IwNDcyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyOVQxNjU5MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZWQyYjdhMTJjNjBlZGQyNzkwNTExYjUzZjA2NzM5OThiMWRmYzczNzk2NDMzM2IwZGM3ZWI0MzQ2YTc1YzJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.w2i-Gv0fKVuIUgca5UqG2z9SQ_Qi2SlPl93JSw5qfkA">
> 
> wasn't too sure about this one: <img alt="image" width="1204" height="390" src="https://private-user-images.githubusercontent.com/200307766/494490726-ff335bc5-c6eb-467c-8bcc-12875037e1a7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkxNjU0NzQsIm5iZiI6MTc1OTE2NTE3NCwicGF0aCI6Ii8yMDAzMDc3NjYvNDk0NDkwNzI2LWZmMzM1YmM1LWM2ZWItNDY3Yy04YmNjLTEyODc1MDM3ZTFhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyOVQxNjU5MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2NhN2NlZmNlMjhmZDQ2MzdhNTIzODBiNjAxZDcwNTljZmFiNmUxNDFlODRjZmYxZDMyZmU3MmRiN2YyNjgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1QNk3MoeC6X156vgdyG5tepH69E40WWMI2B_hMz-zgk">
> 
> **Desktop** <img alt="image" width="876" height="366" src="https://private-user-images.githubusercontent.com/200307766/494492729-30e9c5a4-17fa-412f-a6b6-0f4095402248.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkxNjU0NzQsIm5iZiI6MTc1OTE2NTE3NCwicGF0aCI6Ii8yMDAzMDc3NjYvNDk0NDkyNzI5LTMwZTljNWE0LTE3ZmEtNDEyZi1hNmI2LTBmNDA5NTQwMjI0OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyOVQxNjU5MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OGE1MmFlODhiYjFmNmQ4MDU1NjU2MGRiNGFlNzMwMWQ1N2Q4YmE1OWNiYmZjMmFkZDRiNzg0ZTc4MjkwOTExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qBXVzAX3RT0_2igdM_HFomBdzSHQMhQSj4W5NnF3xJg">

mmmavis commented on foundation_cms/static/scss/mixins.scss:
```diff
@@ -0,0 +1,32 @@
+// ===================================
```
Can we rename this file to `_mixins.scss`, please? My ultimate goal is to have the `_` prefix added to all partial SCSS files that are meant to be imported rather than compiled directly.

mmmavis reviewed (changes requested):
@ramram-mf Thanks for making the updates! I just have one more question & one change request.

ramram-mf commented on foundation_cms/static/scss/mixins.scss:
```diff
@@ -0,0 +1,32 @@
+// ===================================
```
The `text-image-block__narrow` file is a special text-image block that is only used in NP pages, it didn't exist before the NP implementation so markup was implemented rather than changed. This block uses the foundation grid more concretely than the previous `text-image-block` implementation, so classnames were needed in this component, which meant different markup.

nancyt1 reviewed (approved):
Looks amazing!!! Thank you!
I need to give you more feedback on the product cards but not a blocker for this ticket.
